src/kernel/src/fs/dev/dmem1.rs

```diff
@@ -1,18 +1,33 @@
-use crate::errno::Errno;
+use crate::errno::{Errno, EPERM};
 use crate::fs::{IoctlCom, VFile, VFileOps, VPath};
-use crate::process::VThread;
+use crate::process::{VProc, VThread};
 use crate::ucred::Ucred;
 use macros::vpath;
 use std::fmt::{Display, Formatter};
+use std::num::NonZeroI32;
+use std::sync::Arc;
+use thiserror::Error;
 
 #[derive(Debug)]
-pub struct Dmem1 {}
+pub struct Dmem1 {
+    vp: Arc<VProc>,
+    total_size: usize,
+    number: i32,
+}
 
 impl Dmem1 {
     pub const PATH: &VPath = vpath!("/dev/dmem1");
 
-    pub fn new() -> Self {
-        Self {}
+    pub const DMEM_GRP: u8 = 0x80;
+
+    pub const COM10: IoctlCom = IoctlCom::ior::<usize>(Self::DMEM_GRP, 0xa);
+
+    pub fn new(vp: &Arc<VProc>) -> Self {
+        Self {
+            vp: vp.clone(),
+            total_size: 0x180_000_000, // 6 GB
```

Comment:
Where did you get this number from? What I see on the PS4 is it is capped at `0x13C000000`.

Reply:
I got it from fpps4, as I don't know how to get it from the actual PS4

Reply:
![Screenshot from 2023-12-16 16-55-52](https://github.com/obhq/obliteration/assets/3048307/f13e1c8b-3aad-4cb9-a156-c0e717ac4aaa)

Reply:
It's all zeroed for me. Is it initialized somewhere else?

Reply:
Strange. If what you have is a memory dump of the kernel and you load it to Ghidra as a raw memory with correct address it should not be zeroed.

Reply:
I loaded it using the orbis plugin.

Reply:
So, what should I do? Return the 0x13C000000?

Reply:
The correct implementation here is use the value from this dmem device. If you don't know this value then you can use `0x13C000000` with `TODO`.

Reply:
I changed it.